src/app/multi-line-chart/multi-line-chart.component.ts

```diff
@@ -186,19 +186,17 @@ export class MultiLineChartComponent implements OnInit {
         .style('mix-blend-mode', 'multiply')
         .attr('d', (d) => line(d.values as any));
 
-      if (self.animate) {
-        // @ts-ignore
-        path._groups[0].forEach((node, index) => {
-          let length = node.getTotalLength();
-          d3.select(node)
-            .attr('stroke-dasharray', length)
-            .attr('stroke-dashoffset', length)
-            .attr('stroke', VISUALLY_DISTINCT_COLORS[index])
-            .transition()
-            .duration(2000)
-            .attr('stroke-dashoffset', 0);
-        });
-      }
+      // @ts-ignore
+      path._groups[0].forEach((node, index) => {
+        let length = node.getTotalLength();
+        d3.select(node)
+          .attr('stroke-dasharray', length)
+          .attr('stroke-dashoffset', length)
+          .attr('stroke', VISUALLY_DISTINCT_COLORS[index])
+          .transition()
+          .duration(self.animate ? 2000 : 0)
```

Comment:
Easier to read this commit's diff without whitespace: https://github.com/prendradjaja/covid-dashboard/pull/13/commits/0e3dbd09c3696c16388a4798cf382531c747013b?diff=unified&w=1

Bug: Changing `.animate` to `false` causes the data to be invisible instead of just un-animated. Not sure why.

This isn't a good fix. TODO: maybe find a better fix later.

Reply:
that would be because of the stroke-dasharray and stroke-dashoffset, which move the border "outside" of the visible range. (The path is still there, but there's a "dashed border" which rotates visible and non-visible sections into view, creating the illusion of a line animation